lib/ansible/modules/packaging/os/redhat_subscription.py

```diff
@@ -503,14 +505,22 @@ def main():
     consumer_id = module.params["consumer_id"]
     force_register = module.params["force_register"]
 
+    global SUBMAN_CMD
+    SUBMAN_CMD = module.get_bin_path('subscription-manager', False)
+
+    if not SUBMAN_CMD:
+        module.fail_json(msg="Could not find subscription-manager. Please ensure it is installed.")
+
     # Ensure system is registered
     if state == 'present':
 
         # Check for missing parameters ...
-        if not (activationkey or username or password):
-            module.fail_json(msg="Missing arguments, must supply an activationkey (%s) or username (%s) and password (%s)" % (activationkey, username, password))
+        if not (activationkey or org_id or username or password):
```

Comment:
You may be able to simplify this buy using the helper functions
`required_one_of`
`mutually_exclusive`
`required_together`

Have a look at existing modules for examples